spi/control-plane/contract-spi/src/testFixtures/java/org/eclipse/edc/connector/contract/spi/testfixtures/offer/store/ContractDefinitionStoreTestBase.java

```diff
@@ -488,6 +488,67 @@ void shouldReturnAll_with_private_properties_whenNoFiltersApplied() {
             var definitionsRetrieved = getContractDefinitionStore().findAll(QuerySpec.max());
             assertThat(definitionsRetrieved).isNotNull().hasSize(2);
         }
+
+        @Test
+        void shouldReturn_with_private_propertiesFilter() {
+            var definition1 = createContractDefinition("definition1", "policyId", "contractId", Map.of("key1", "value1"));
+            getContractDefinitionStore().save(definition1);
+            var definition2 = createContractDefinition("definition2", "policyId", "contractId", Map.of("key2", "value2"));
+            getContractDefinitionStore().save(definition2);
+
+
+            var spec = QuerySpec.Builder.newInstance()
+                    .filter(new Criterion("privateProperties.key1", "=", "value1"))
+                    .build();
+
+            var definitionsRetrieved = getContractDefinitionStore().findAll(spec);
+            assertThat(definitionsRetrieved).isNotNull().hasSize(1);
+
+            spec = QuerySpec.Builder.newInstance()
+                    .filter(new Criterion("privateProperties.key2", "=", "value2"))
+                    .build();
+
+            definitionsRetrieved = getContractDefinitionStore().findAll(spec);
+            assertThat(definitionsRetrieved).isNotNull().hasSize(1);
```

Comment:
Shouldn't this verify the correct definition is returned?

Reply:
done